server/routes/getSlashes.ts

```diff
@@ -0,0 +1,57 @@
+import Sequelize from 'sequelize';
+import { Request, Response, NextFunction } from 'express';
+import { Errors } from './getOffences';
+
+const Op = Sequelize.Op;
+
+const slashQuery = async (models, req: Request, res: Response, next: NextFunction) => {
+    const { chain, stash } = req.query;
+    let { startDate, endDate } = req.query;
+
+    console.log("=========== slashes ========= ", chain, stash)
+
+    if (!chain) return next(new Error(Errors.ChainIdNotFound));
+
+    const chainInfo = await models.Chain.findOne({ where: { id: chain } });
+
+    if (!chainInfo) return next(new Error(Errors.InvalidChain));
+
+    // if date isn't defined we get for last 30 days
+    if (typeof startDate === 'undefined' || typeof endDate === 'undefined') {
+        const daysAgo = 30;
+        startDate = new Date(); // today
+        startDate.setDate(startDate.getDate() - daysAgo); // 30 days ago
+        startDate = new Date(startDate); // formatted
+        endDate = new Date(); // today
+    }
+
+    const where: any = {
```

Comment:
@jnaviask is this the proper way to query for chain events? I'm unsure with it mixing the ChainEventType model and searching over ChainEvents?

Reply:
Yes this seems fine, every ChainEvent should have a ChainEventType associated with it, so as long as we query with the include it _should_ work. Usually, however, I query the ChainEventTypes we need first, and then pass their `id`s into the ChainEvent query.